### PR TITLE
Smtp service

### PR DESCRIPTION
**Pull request by brettminnie**

### CODE REVIEW

File: service/smtp-service.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+      }
+    )
+  );
+};
+
+/**
+ * Load an email template specified in the body of the model
+ */
+SmtpService.prototype.loadTemplate = function loadTemplate() {
+  return fs.readFileAsync(this.dataModel.template).then(function bindTemplate(data) {
```
Comment (JoeChapman) [inline service/smtp-service.js]: use a getter instead of `this.dataModel.template`

File: service/smtp-service.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  });
+};
+
+/**
+ * Work out magic and return the processed template
+ * @returns {*|String}
+ */
+SmtpService.prototype.prepareContent = function prepareContent() {
+
+  return fs.readFileAsync(this.dataModel.template).then(function bindData(data) {
```
Comment (JoeChapman) [inline service/smtp-service.js]: use a getter instead of `this.dataModel.template`

File: service/smtp-service.js (hunk 1 above)
Comment (brettminnie) [inline service/smtp-service.js]: cool, will wait for all your comments to implement this change

File: service/smtp-service.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    this.template = data;
+    this.compiledTemplate = templateEngine.compile(this.template.toString());
```
Comment (JoeChapman) [inline service/smtp-service.js]: Is this  - `this.template.toString()` - how Handlebars proposes we 'stringify' our templates? it feels like there should be an API method for rendering strings.

File: service/smtp-service.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+  var mailOptions = {
+    from: this.dataModel.sender,
+    to: this.dataModel.recipient,
+    subject: this.dataModel.subject,
+    html: this.compiledTemplate,
+    generateTextFromHTML: true,
+  };
+
+  return new Promise(function sendMailAysnc(res, rej) {
```
Comment (JoeChapman) [inline service/smtp-service.js]: could you use the long-handed version of the arguments `res` and `rej` so as not to confuse `res` with response?

Comment (brettminnie) [inline service/smtp-service.js]: :+1:

File: service/smtp-service.js (hunk 3 above)
Comment (brettminnie) [inline service/smtp-service.js]: `compile` expects a string `fs` returns a bytestream, changed the casting so it does it prior to pushing it into the compile `method`

**Pull request merged**